frontend/src/components/popup/Settings.tsx

```diff
@@ -228,6 +230,69 @@ const SectionConflictSettings = memo(function SectionConflictSettings() {
     );
 });
 
+const ExperimentalFeaturesSettings = memo(
+    function ExperimentalFeaturesSettings() {
+        const experimentalFeaturesOptions = useStore(
+            (store) => store.experimentalFeaturesOptions,
+        );
+        const setExperimentalFeaturesOptions = useStore(
+            (store) => store.setExperimentalFeaturesOptions,
+        );
+        const rangeOptions = [2, 4, 6, 8, 10, "all"];
+
+        return (
+            <div className={Css.experimentalFeatures}>
+                <h3 className={Css.title}>Experimental Features</h3>
+                <span>
+                    Search for matching courses from recent terms when no course
+                    is found
+                </span>
+
+                <Slider
+                    value={experimentalFeaturesOptions.enableHistoricalSearch}
+                    onToggle={() => {
+                        setExperimentalFeaturesOptions({
+                            ...experimentalFeaturesOptions,
+                            enableHistoricalSearch:
+                                !experimentalFeaturesOptions.enableHistoricalSearch,
+                        });
+                    }}
+                    text=""
+                />
+                <span>
+                    How many recent terms do you want to search?{" "}
+                    <span className={Css.warning}>
+                        {" "}
+                        (this may affect performance!)
+                    </span>
+                </span>
+                <Dropdown
+                    choices={rangeOptions}
+                    selected={
+                        experimentalFeaturesOptions.historicalSearchRange !==
+                        Infinity
+                            ? experimentalFeaturesOptions.historicalSearchRange
+                            : "all"
+                    }
+                    onSelect={(index) => {
+                        let range: number;
+                        if (rangeOptions[index] === "all") {
+                            range = Infinity;
+                        } else {
+                            range = rangeOptions[index] as number;
+                        }
+                        setExperimentalFeaturesOptions({
+                            ...experimentalFeaturesOptions,
+                            historicalSearchRange: range,
+                        });
+                    }}
+                    emptyPlaceholder={NaN}
```

Comment:
Same as above but `NaN` is even worse. The `emptyPlaceholder` here is a message and should be a string to tell user nothing is selected.

Reply:
I will fix this part to make it makes more sense!